helpdesk_mgmt/tests/test_portal.py

```diff
@@ -0,0 +1,124 @@
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn("portal-ticket-title", resp.text)
+        self.assertIn('<h4 class="page-header">History</h4>', resp.text)
+
+    def test_submit_ticket(self):
+        """Submit a ticket in portal mode."""
+
+        new_ticket_title = "portal-new-submitted-ticket-subject"
+
+        def count_new_tickets():
+            return self.env["helpdesk.ticket"].search_count(
+                [("name", "=", new_ticket_title)]
+            )
+
+        self.assertEqual(count_new_tickets(), 0)  # not found so far
+
+        self.authenticate("portal", "portal")
+        resp = self.url_open(
+            "/submitted/ticket",
+            data={
+                "category": self.env.ref("helpdesk_mgmt.helpdesk_category_1").id,
+                "csrf_token": http.WebRequest.csrf_token(self),
+                "subject": new_ticket_title,
+                "description": "portal-new-submitted-ticket-description",
+            },
+        )
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(count_new_tickets(), 1)  # new ticket found
+
+    def test_ticket_list_unauthenticated(self):
+        """Attempt to list tickets without auth, ensure we get sent back to login."""
+
+        resp = self.url_open("/my/tickets", allow_redirects=False)
+        self.assertEqual(resp.status_code, 303)
+        self.assertTrue(resp.is_redirect)
+        # http://127.0.0.1:8069/web/login?redirect=http%3A%2F%2F127.0.0.1%3A8069%2Fmy%2Ftickets
+        self.assertIn("/web/login", resp.headers["Location"])
+
+    def test_ticket_list_unauthorized(self):
+        """Attempt to list tickets without ticket ACL, ensure we get sent back
+        to /my.
+        """
+
+        self.env.ref("helpdesk_mgmt.access_helpdesk_ticket_portal").perm_read = False
+        self.authenticate("portal", "portal")
+        resp = self.url_open("/my/tickets", allow_redirects=False)
+        self.assertEqual(resp.status_code, 303)
+        self.assertTrue(resp.is_redirect)
+        # http://127.0.0.1:8069/my
+        self.assertTrue(resp.headers["Location"].endswith("/my"))
+
+    def test_tickets_2_users(self):
+        """Check tickets between 2 portal users; ensure they can't access each
+        others' tickets.
+        """
+
+        portal_user_1 = self.user_portal  # created by HttpCaseWithUserPortal
+        portal_user_2 = self.user_portal.copy(
+            default={"login": "portal2", "password": "portal2"}
+        )
+
+        self._create_ticket(portal_user_1.partner_id, "ticket-user-1")
+        self._create_ticket(portal_user_2.partner_id, "ticket-user-2")
+
+        # portal_user_1 only sees ticket_1
+        self.authenticate("portal", "portal")
+        resp = self.url_open("/my/tickets")
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn("ticket-user-1", resp.text)
+        self.assertNotIn("ticket-user-2", resp.text)
```

Comment:
```suggestion
        self.assertNotIn("ticket-user-2", resp.text)
        resp = self.url_open(f"/my/ticket/{ticket_1.id}")
        self.assertEqual(resp.status_code, 200)
        self.assertIn("ticket-user-1", resp.text)
        resp = self.url_open(f"/my/ticket/{ticket_2.id}")
        self.assertEqual(resp.status_code, 301)
```
And something similar to portal2 user.

Reply:
Ok yes I was testing access rule via ticket list but not ticket form, will add